python/python/tests/test_optimize.py

```diff
@@ -123,6 +126,41 @@ def check_index(has_knn_combined):
     check_index(has_knn_combined=True)
 
 
+def test_index_remapping_multiple_rewrite_tasks(tmp_path: Path):
+    base_dir = tmp_path / "dataset"
+    ds = lance.write_dataset(
+        create_table(min=0, max=1, nvec=300), base_dir, mode="overwrite"
+    )
+    ds = ds.create_index(
+        "vector",
+        index_type="IVF_PQ",
+        num_partitions=4,
+        num_sub_vectors=2,
+    )
+
+    assert ds.has_index
+
+    tbl = create_table(min=0, max=1, nvec=200)
+    ds = lance.write_dataset(tbl, base_dir, mode="append")
+
+    ds.delete("id % 4 == 0")
+    # We have a dataset with 2 small fragments.  Both fragments will
+    # be compacted with themselves since they have many deleted rows.
+    #
+    # However, they should not be compacted together because one of
+    # them is indexed and the other is not.
```

Comment:
Curious what would that look llike if we have many indices on different columns created from many version

Reply:
I believe the planner will not combine two fragments unless they are covered by the same indices.  So if you have:

Index A: [0, 1, 2, 3]
Index B: [1, 2, 3]
Index C: [3]

Then you will get: [0], [1, 2], [3]